php/Terminus/Models/Environment.php

```diff
@@ -432,6 +448,42 @@ public function getName() {
   }
 
   /**
+   * Returns the parent environment
+   *
+   * @return Environment
+   */
+  public function getParentEnvironment() {
+    $env_id = $this->get('id');
+    if ($env_id == 'dev') {
+      return null;
+    }
+    switch ($this->get('id')) {
+      case 'dev':
+          return null;
+          break;
+      case 'live':
+        $parent_env_id = 'test';
+          break;
+      default:
+        $parent_env_id = 'dev';
+          break;
+    }
+    $environment = $this->site->environments->get($parent_env_id);
+    return $environment;
+  }
+
+  /**
+   * Decides if the environment has changes to deploy
+   *
+   * @return bool
+   */
+  public function hasUpstreamUpdates() {
```

Comment:
Let's not call this Upstream Updates. That very specifically means "does the parent repo have commits we need to pull down?" and not confuse it with "is the environment at a different head than another environment?" Maybe "hasDeployableCode"?